File: Electronics_Shop_17.Model/Requests/AddDiscount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddDiscount
    {
        public string DiscountType { get; set; }
        public int Amount { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsActive { get; set; }        

        public virtual List<AddProductDiscount> ProductDiscounts { get; set; }
    }
}

File: angular17_web/src/app/models/discount/add.discount.ts
import { AddProductDiscount } from '../product-discount/add.product.discount';

export class AddDiscount {
  discountType!: string;
  amount!: number;
  startDate!: Date;
  endDate!: Date;
  isActive!: boolean;
  productDiscounts: AddProductDiscount[] = [];

  constructor(data?: any) {
    if (!data) return;

    this.discountType = data.discountType;
    this.amount = data.amount;
    this.startDate = data.startDate ? new Date(data.startDate) : undefined as any;
    this.endDate = data.endDate ? new Date(data.endDate) : undefined as any;
    this.isActive = data.isActive;
    this.productDiscounts = Array.isArray(data.productDiscounts)
      ? data.productDiscounts.map((x: any) => new AddProductDiscount(x))
      : [];
  }
}
